exporter/loadbalancingexporter/loadbalancer_test.go

```diff
@@ -111,6 +111,30 @@ func TestWithDNSResolver(t *testing.T) {
 	assert.True(t, ok)
 }
 
+func TestWithDNSResolverNoEndpoints(t *testing.T) {
+	cfg := &Config{
+		Resolver: ResolverSettings{
+			DNS: &DNSResolver{
+				Hostname: "service-1",
+			},
+		},
+	}
+	p, err := newLoadBalancer(componenttest.NewNopExporterCreateSettings(), cfg, nil)
+	require.NotNil(t, p)
+	require.NoError(t, err)
+
+	// test
+	res, ok := p.res.(*dnsResolver)
```

Comment:
This line isn't the actual test. What you are testing in this unit test is the call to p.Endpoint.
You also don't need this type assertion here, but you do need to call Start, to make the test semantically correct. Here's a revised version of the test:

```go
func TestWithDNSResolverNoEndpoints(t *testing.T) {
	// prepare
	cfg := &Config{
		Resolver: ResolverSettings{
			DNS: &DNSResolver{
				Hostname: "service-1",
			},
		},
	}
	p, err := newLoadBalancer(componenttest.NewNopExporterCreateSettings(), cfg, nil)
	require.NotNil(t, p)
	require.NoError(t, err)

	err = p.Start(context.Background(), componenttest.NewNopHost())
	require.NoError(t, err)

	// test
	e := p.Endpoint(pcommon.NewTraceID([16]byte{128, 128, 0, 0}))

	// verify
	assert.Equal(t, "", e)
}
```

Reply:
Thanks. I'll use your test. FWIW I just copied from the test above (monkey see, monkey do).